Utilities/simple_ringbuf.h

```diff
@@ -4,35 +4,53 @@
 #include "util/atomic.hpp"
 
 // Single reader/writer simple ringbuffer.
-// Counters are 32-bit.
 class simple_ringbuf
 {
-private:
-
-	atomic_t<u64> rw_ptr = 0;
-	u32 buf_size = 0;
-	std::unique_ptr<u8[]> buf{};
-	atomic_t<bool> initialized = false;
-
 public:
 
-	simple_ringbuf() {};
-	simple_ringbuf(u32 size);
+	using ctr_t = u64;
```

Comment:
Could have used a generic C++ type std::size_t which is defined as `usz` in rpcs3.